File: benchmarks/programs/strings.py
# String formatting + concatenation churn. %-format / join / len.
parts = []
acc = 0
for i in range(1, 200001):
    s = "item-%d-%x" % (i, i * 7)
    parts.append(s)
    acc += len(s)
big = ",".join(parts)
print("%d" % ((acc + len(big)) % 1000000007))

File: benchmarks/programs/strings.lua
-- String formatting + concatenation churn. string.format / table.concat / #.
local parts = {}
local acc = 0
for i = 1, 200000 do
  local s = string.format("item-%d-%x", i, i * 7)
  parts[#parts + 1] = s
  acc = acc + #s
end
local big = table.concat(parts, ",")
print(string.format("%d", (acc + #big) % 1000000007))
